.github/workflows/ros-ci.yaml

```diff
@@ -0,0 +1,64 @@
+name: ros-ci
+
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    branches:
+      - main
+
+jobs:
+  asan:
+    strategy:
+      fail-fast: false
+      matrix:
+        env:
+          - ROS_DISTRO: rolling
+          - ROS_DISTRO: humble
+
+    name: ${{ matrix.env.ROS_DISTRO }}-asan
+    runs-on: ubuntu-22.04
```

Comment:
I'm glad we're using tagged runners. It makes it easier to bisect problems when we know the actual OS we were using at any given point in time.